File: 14.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int count = 0 , min = 201,len;
        string str;
        for(int i = 0;i < strs.size() ; i++)
        {
            len = strs[i].length();
            if(len<min)
            {
                min = len;
                str = strs[i];
            }
        }
        for(int i=0 ; i<str.length();i++)
        {
            for(int j=0;j<strs.size();j++)
            {
                if(strs[j][i] != str[i])
                    return str.substr(0,count);
            }
            count++;
        }
        return str.substr(0,count);

    }
};

File: 14.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        count = int(0)
        first = strs[0]
        last = strs[-1]
        for i in range(min(len(first),len(last))):
            if first[i] != last[i]:
                if count == 0:
                    return ''
                return first[0:count]
            count+=1
        if count == 0:
            return ''
        return first[0:count]